lib/modules/dosomething/dosomething_kudos/dosomething_kudos.module

```diff
@@ -131,67 +126,105 @@ function dosomething_kudos_sort_by_taxonomy_term($a, $b) {
   return $result;
 }
 
-
 /**
- * Group all kudos in collection and organize them by taxonomy term.
+ * Group all kudos in collection and organize them by taxonomy term,
+ * including current users activity on retrieved kudos records.
  *
  * @param  array $data Array containing Kudos items.
- * @param  array $options
  * @return array
  */
-function dosomething_kudos_group_by_taxonomy_term($data, $options = []) {
-  $terms = [];
+function dosomething_kudos_group_by_taxonomy_term($data) {
+  $output = [];
+  $activity = [];
 
   foreach ($data as $item) {
     if (!($item instanceof Kudos)) {
       $item = (object) $item;
     }
 
-    if (!array_key_exists($item->term['id'], $terms)) {
-      $terms[$item->term['id']] = [
+    // If term is not in approved reactions list, skip it!
+    if (!in_array($item->term['name'], Kudos::getApprovedReactions())) {
+      continue;
+    }
+
+    // If the term is not already in the $output array then add it.
+    if (!array_key_exists($item->term['id'], $output)) {
+      $output[$item->term['id']] = [
         'term' => $item->term,
       ];
     }
 
-    $output = [
-      'id' => $item->id,
-      'user' => $item->user,
-    ];
-
-    if ($options['display'] === 'full') {
-      $output['reportback_item'] = $item->reportback_item;
+    if (!isset($output[$item->term['id']]['term']['total'])) {
+      $output[$item->term['id']]['term']['total'] = 0;
     }
 
-    if (isset($item->uri)) {
-      $output['uri'] = $item->uri;
-    }
+    $output[$item->term['id']]['term']['total'] += 1;
```

Comment:
Defs agree with you though. The aim for this was to ultimately help with efficiency in what gets sent back in a JSON response so the kudos object isn't crazy long with like 900 things. SQL query performance would be a next step :)